exercises/02-publish-subscribe/after/Divergent.Finance/Host.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Threading.Tasks;
+using Divergent.Finance.PaymentClient;
+using Divergent.Sales.Messages.Events;
+using ITOps.EndpointConfig;
+using NServiceBus;
+using NServiceBus.Logging;
+
+namespace Divergent.Finance
+{
+    class Host
+    {
+        static readonly ILog Log = LogManager.GetLogger<Host>();
+        readonly string connectionString;
+        IEndpointInstance endpoint;
+
+        public static string EndpointName => "Divergent.Finance";
+
+        public Host(string connectionString) => this.connectionString = connectionString;
+
+        public async Task Start()
+        {
+            try
+            {
+                var endpointConfiguration = new EndpointConfiguration(EndpointName)
```

Comment:
Seperate components are created so responsibilities aren't mixed, but then this mixes endpointconfig with the host and breaks SRP again?

Reply:
"Single responsibility" is usually a matter of opinion. 😉 

I'd argue that we know have a top level entry point (Program.cs), a windows service adapter (WindowService.cs) and a host for the endpoint (Host.cs). But in any case, this is following the pattern of our templates, so I'd prefer if we could merge as-is (to solve the pressing issue of using NSB.Host) and we address these concerns in the templates. If the templates are changed then we can carry those changes over into the exercises.